### PR TITLE
Clear txs on resync and remove wallet

### PR DESCRIPTION
**Pull request by banklesss**

### CODE REVIEW

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 1]
```diff
@@ -78,6 +78,10 @@ export class TransactionCache {
     })
   }
 
+  clearTxs() {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts]: clear

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 2]
```diff
@@ -465,6 +469,7 @@ type SyncMetadata = {
 export type TxCacheStorage = {
   loadTxs: () => Promise<Record<string, Transaction>>
   saveTxs: (txs: Record<string, Transaction>) => Promise<void>
+  removeTxs: () => Promise<void>
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts]: clear

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 3]
```diff
@@ -496,6 +501,11 @@ const makeTxCacheStorage = (storage: Storage): TxCacheStorage => ({
       storage.setItem('txids', JSON.stringify(keys)),
     ])
   },
+
+  removeTxs: async () => {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts]: clear

File: src/yoroi-wallets/cardano/ShelleyWallet.ts [hunk 4]
```diff
@@ -95,8 +95,9 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
     return this.storage.write(`/wallet/${this.id}/data`, this.toJSON())
   }
 
-  clear() {
-    return this.clearUTXOs()
+  async clear() {
+    await this.transactionCache?.clearTxs()
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/ShelleyWallet.ts]: .clear()

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts [hunk 5]
```diff
@@ -496,6 +503,11 @@ const makeTxCacheStorage = (storage: Storage): TxCacheStorage => ({
       storage.setItem('txids', JSON.stringify(keys)),
     ])
   },
+
+  clearTxs: async () => {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/shelley/transactionCache.ts]: ~`clearTxs`~ -> `clear`

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts (hunk 1 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/shelley/transactionCache.ts]: ~`clearTxs`~ -> `clear`

File: src/yoroi-wallets/cardano/ShelleyWallet.ts (hunk 4 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/ShelleyWallet.ts]: ~`clearTxs`~ -> `clear`

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts (hunk 2 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/shelley/transactionCache.ts]: ~`removeTxs`~ -> `clear`

File: src/yoroi-wallets/cardano/shelley/transactionCache.ts (hunk 3 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/shelley/transactionCache.ts]: ~`removeTxs`~ -> `clear`

File: src/yoroi-wallets/storage.ts [hunk 6]
```diff
@@ -23,5 +25,11 @@ export const makeStorageWithPrefix = (prefix: string): Storage => {
     multiSet: (items: Array<[string, string]>) => {
       return AsyncStorage.multiSet(items.map(([key, value]) => [withPrefix(key), value]))
     },
+    removeItem: (key: string) => {
+      return AsyncStorage.removeItem(withPrefix(key))
+    },
+    multiRemove: (keys: Array<string>) => {
+      return AsyncStorage.multiRemove(keys.map((key) => withPrefix(key)))
+    },
```
Comment (wolverineks) [inline src/yoroi-wallets/storage.ts:38]: something like this, and make sure it works
```
clear: async () => {
  const keys = await AsyncStorage.getAllKeys()
  const filteredKeys = keys.filter((key) => key.startsWith(prefix))
   
  return AsyncStorage.multiRemove(filteredKeys)
}
```

File: src/yoroi-wallets/storage.ts [hunk 7]
```diff
@@ -23,5 +25,11 @@ export const makeStorageWithPrefix = (prefix: string): Storage => {
     multiSet: (items: Array<[string, string]>) => {
       return AsyncStorage.multiSet(items.map(([key, value]) => [withPrefix(key), value]))
     },
+    removeItem: (key: string) => {
```
Comment (wolverineks) [inline src/yoroi-wallets/storage.ts:33]: keep these

**Review by wolverineks — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

File: src/yoroi-wallets/cardano/ShelleyWallet.ts [hunk 8]
```diff
@@ -95,8 +95,9 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
     return this.storage.write(`/wallet/${this.id}/data`, this.toJSON())
   }
 
-  clear() {
-    return this.clearUTXOs()
+  async clear() {
+    await this.transactionCache?.clear()
+    await this.clearUTXOs()
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/ShelleyWallet.ts:100]: ~`this.clearUTXOs()`~ -> `this.utxoStorage.clearUtxoState()`

**Pull request merged**